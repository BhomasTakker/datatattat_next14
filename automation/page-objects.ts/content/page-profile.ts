import { Locator, Page } from "@playwright/test";
import { ContentPage } from "./content-page";

// argument for content being a composite of sub classes
// pass page/content in and initialize each sub class
export class PageProfile extends ContentPage {
	private readonly _profile: Locator;

	constructor(page: Page) {
		super(page);
		this._profile = this.page.getByTestId("page-profile-root");
	}

	get profile(): Locator {
		return this._profile;
	}

	getContentTitle() {
		return this.profile.getByRole("heading", { name: "Automation Page" });
	}
}
