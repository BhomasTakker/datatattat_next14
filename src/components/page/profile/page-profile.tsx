import { PageProfile as IPageProfile } from "@/types/page";
import styles from "./page-profile.module.scss";

interface PageProfileProps {
	profile: IPageProfile;
}

// Show title
// Eventually show description, createdBy, created, updated, etc
export const PageProfile = ({ profile }: PageProfileProps) => {
	const { pageTitle, showPageTitle } = profile || {};

	return (
		// set classname to be root & root-{given-variant}
		<div className={styles.root} data-testid="page-profile-root">
			{showPageTitle && pageTitle ? (
				<h1 className={styles.title}>{pageTitle}</h1>
			) : null}
		</div>
	);
};
