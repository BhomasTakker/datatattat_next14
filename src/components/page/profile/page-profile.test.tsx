import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { PageProfile } from "./page-profile";

describe("PageProfile Component", () => {
	it("should render without crashing", () => {
		const { container } = render(<PageProfile profile={{}} />);
		expect(container).toBeInTheDocument();
	});

	it("should not render the title if showPageTitle is false", () => {
		render(
			<PageProfile
				profile={{ pageTitle: "Test Title", showPageTitle: false }}
			/>
		);
		expect(screen.queryByText("Test Title")).not.toBeInTheDocument();
	});

	it("should not render the title if pageTitle is not provided", () => {
		render(<PageProfile profile={{ showPageTitle: true }} />);
		expect(screen.queryByText("Test Title")).not.toBeInTheDocument();
	});

	it("should render the title if showPageTitle is true and pageTitle is provided", () => {
		render(
			<PageProfile profile={{ pageTitle: "Test Title", showPageTitle: true }} />
		);
		expect(screen.getByText("Test Title")).toBeInTheDocument();
	});

	it("should apply the correct styles", () => {
		render(
			<PageProfile profile={{ pageTitle: "Test Title", showPageTitle: true }} />
		);

		// Should not resort to using test ids....
		expect(screen.getByTestId("page-profile-root")).toHaveClass("root");
		expect(screen.getByRole("heading")).toHaveClass("title");
	});

	it("renders page profile unchanged", () => {
		const { container } = render(
			<PageProfile profile={{ pageTitle: "Test Title", showPageTitle: true }} />
		);
		expect(container).toMatchSnapshot();
	});
});
