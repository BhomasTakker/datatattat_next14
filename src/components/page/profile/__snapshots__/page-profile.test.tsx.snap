// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`PageProfile Component renders page profile unchanged 1`] = `
<div>
  <div
    class="root"
    data-testid="page-profile-root"
  >
    <h1
      class="title"
    >
      Test Title
    </h1>
  </div>
</div>
`;
